lib/commands/pull.sh

```diff
@@ -35,13 +46,15 @@ symlink_new_files() {
     local castle=$1
     shift
     local repo="$repos/$castle"
+    local before_pull
+    if ! before_pull=$(cd "$repo" && git rev-parse "refs/tags/$BEFORE_PULL_TAG" && git tag -d "$BEFORE_PULL_TAG" >/dev/null); then
+      continue
+    fi
     if [[ ! -d $repo/home ]]; then
       continue;
     fi
     local git_out
-    local now
-    now=$(date +%s)
-    if ! git_out=$(cd "$repo" && git diff --name-only --diff-filter=A "HEAD@{(($now-$T_START+1)).seconds.ago}" HEAD -- home 2>/dev/null | wc -l 2>&1); then
+    if ! git_out=$(cd "$repo" && git diff --name-only --diff-filter=AR "$before_pull" HEAD -- home 2>/dev/null | wc -l 2>&1); then
```

Comment:
Why not just diff to the tag and then delete it after? It's one less command that can go wrong.

Reply:
I think it was to make sure the tag still gets deleted in case `$repo/home` isn't a directory, but it looks like we can just move that check after the diff.